src/autorest-core/language-service/language-service.ts

```diff
@@ -0,0 +1,653 @@
+    return docAnalysis ? <Hover>{
+      contents: [
+        ...this.onHoverRef(docAnalysis, position.position),
+        ...this.onHoverJsonPath(docAnalysis, position.position)
+      ]
+    } : <Hover><any>null
+  }
+
+  private onDefinitionRef(docAnalysis: DocumentAnalysis, position: Position): Iterable<Location> {
+    const refValueJsonPath = docAnalysis.GetJsonPathFromJsonReferenceAt(position);
+    if (refValueJsonPath) {
+      return docAnalysis.GetDocumentLocations(refValueJsonPath);
+    } // else  { console.log("found nothing that looks like a JSON reference"); }
+    return [];
+  }
+
+  private onDefinitionJsonPath(docAnalysis: DocumentAnalysis, position: Position): Iterable<Location> {
+    const potentialQuery: string = <string>docAnalysis.GetJsonQueryAt(position);
+    if (potentialQuery) {
+
+      return docAnalysis.GetDocumentLocations(potentialQuery);
+    } // else  { console.log("found nothing that looks like a JSON path");}
+    return [];
+  }
+
+  private async onDefinition(position: TextDocumentPositionParams): Promise<Location[]> {
+    const docAnalysis = await this.getDocumentAnalysis(position.textDocument.uri);
+    return docAnalysis ? [
+      ...this.onDefinitionRef(docAnalysis, position.position),
+      ...this.onDefinitionJsonPath(docAnalysis, position.position)
+    ] : [];
+  }
+
+  private async onFileEvents(changes: FileEvent[]) {
+    this.debug(`onFileEvents: ${changes}`);
+    for (const each of changes) {
+
+      const doc = this.get(each.uri);
+      if (doc) {
+        this.onDocumentChanged(doc);
+        return;
+      }
+
+      let documentUri = each.uri;
+      const txt = await this.ReadFile(each.uri);
+      if (documentUri.startsWith("file://")) {
+        // fake out a document for us to play with
+        this.onDocumentChanged({
+          uri: each.uri,
+          languageId: "",
+          version: 1,
+          getText: () => txt,
+          positionAt: (offset: number) => <Position>{},
+          offsetAt: (position: Position) => 0,
+          lineCount: 1
+        });
+      }
+    }
+  }
+  // IFileSystem Implementation
+  public async EnumerateFileUris(folderUri: string): Promise<Array<string>> {
+    if (folderUri && folderUri.startsWith("file:")) {
+      const folderPath = FileUriToPath(folderUri);
+      if (await isDirectory(folderPath)) {
+        const items = await readdir(folderPath);
+        return From<string>(items).Where(each => AutoRest.IsConfigurationExtension(GetExtension(each))).Select(each => ResolveUri(folderUri, each)).ToArray();
+      }
+    }
+
+    return [];
+  }
+
+  public async ReadFile(fileUri: string): Promise<string> {
+    const doc = this.get(fileUri) || this.virtualFile.get(fileUri);
+    try {
+      if (doc) {
+        return doc.getText();
+      }
+
+      const content = await readFile(FileUriToPath(fileUri));
+
+      return content;
+    } catch {
+    }
+    return "";
+  }
+
+  private async process(configurationUrl: string) {
```

Comment:
could you add comments on what this does? It appears to be the main entry point to get the output of AutoRest, basically "executeAutorest"